### PR TITLE
Generic show method for registering commands.

### PR DESCRIPTION
**Pull request by williexu**
---
This will fail with exit code of 3 if resource is missing.

This will allow resolving of https://github.com/Azure/azure-cli/issues/5006, once this is in, I will go ahead and register all show commands in this new way.

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR has modified HISTORY.rst describing any customer-facing, functional changes. Note that this does not include changes only to help content. (see [Modifying change log](https://github.com/Azure/azure-cli/tree/master/doc/authoring_command_modules#modify-change-log)).

### CODE REVIEW

File: src/azure-cli-core/azure/cli/core/commands/arm.py [hunk 1]
```diff
@@ -617,6 +618,48 @@ def handler(args):
...
+
+    factory = _get_client_factory(name, kwargs)
+
+    def generic_show_arguments_loader():
+
+        getter_args = dict(extract_args_from_signature(context.get_op_handler(getter_op),
+                                                       excluded_params=EXCLUDED_PARAMS))
+        cmd_args = getter_args.copy()
+        cmd_args['cmd'] = CLICommandArgument('cmd', arg_type=ignore_type)
+        return [(k, v) for k, v in cmd_args.items()]
```
Comment (tjprescott) [inline src/azure-cli-core/azure/cli/core/commands/arm.py]: Is this the same code as generic wait/update?  If so we could probably extract these to a common implementation for maintainability.

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/commands.py [hunk 2]
```diff
@@ -44,7 +44,7 @@ def get_custom_sdk(custom_module, client_factory, resource_type=ResourceType.DAT
         g.command('check-name', 'check_name_availability')
         g.custom_command('create', 'create_storage_account', min_api='2016-01-01')
         g.command('delete', 'delete', confirmation=True)
-        g.command('show', 'get_properties', exception_handler=g.get_handler_suppress_404())
+        g.generic_show_command('show', 'get_properties')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/commands.py]: Does this represent a breaking change to this command?  If we are including this on one command as a test, it should ensure that the behavior is unchanged.  We don't want to go from swallowing a 404 to failing.  Better to go from failing to failing.  We would need to advertise the behavior change.

Comment (williexu) [reply src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/commands.py]: Nope, the `g.get_handler_suppress_404()` only handles data-plane storage 404 exceptions. 
This is an example of an exception handler that was added but never did its intended job.

This won't break any scripts unless the user checked specifically for an exit code of 1.

Comment (tjprescott) [reply src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/commands.py]: What would they have previously looked for?

**Comment by tjprescott** [comment]
Also, do note that merging this would not permit resolving #5006 since now, if anything, you've actually made the show behavior more inconsistent.  You will need a strategy for how to actually convert over the existing commands, which will involve informing partner teams if you plan to impact their commands (and thus their customers).  Review your plan with @mayurid

File: src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/commands.py (hunk 2 above)
Comment (williexu) [reply src/command_modules/azure-cli-storage/azure/cli/command_modules/storage/commands.py]: sample script from #5006 : 
```az group show $resourceGroupName 1> /dev/null

if [ $? != 0 ]; then
...
fi
```
Most script will check for exit code not being 0 or the presence of output, as error codes can vary.

**Comment by troydai** [comment]
Resolve the conflict please.

**Comment by tjprescott** [comment]
I would recommend not merging this in this sprint and instead merging it right after the release.

**Pull request merged**